docs/reference/migration/migrate_7_0/migrate_to_java_time.asciidoc

```diff
@@ -0,0 +1,350 @@
+[[migrate_to_java_time]]
+=== Java time migration guide
+
+With 7.0, {es} switch to java time from joda time for date-related parsing,
+formatting, and calculations. This guide is designed to help you determine
+if your cluster is impacted and, if so, prepare for the upgrade.
+
+Joda style indices are supported in 6.8, but indices created with joda-style patterns in 6.x are also allowed to be used in 7.0
+All new indices created in 7.x are expected to be using java-time patterns.
+
+We advice to prepare for this migration in 6.8, but it will also work if you decide to perform it in version 7.x
+
+- In 6.8 you need to prefix your new java.style pattern with `8`. Patterns without `8` are considered joda style.
+- In 7.x patterns without `8` on indices *created in 6.x* are considered joda-style.
+- Patterns with or without `8` on indices *created in 7.x*  are considered java.style
```

Comment:
I would add this as a new section. If possible, move it below the "Impacted features" section.

```suggestion
[[prefix-java-time-date-formats]]
=== Prefix for java-time date formats

To upgrade to 7.0, you'll need to convert any impacted joda-time date patterns
to their java-time equivalents.

To aid this migration, you can prefix java-time date patterns with an `8` prefix
in {es} 6.8 and later versions.

For example, you can change the date format `YYYY-MM-dd` to `8YYYY-MM-dd` to
indicate the date format uses java-time.

How {es} handles date formats using the `8` prefix depends on the version:

*6.8*: Date formats with an `8` prefix are handled as java-time formats. Date
formats without an `8` prefix are handled as joda-time formats. We recommend
converting these joda-time formats to java-time _before_ upgrading to 7.x.

*7.x*: For indices created in 6.x, date formats without an `8` prefix are
handled as joda-time formats. For indices created in 7.x, all date formats are
considered java-style, regardless of an `8` prefix.
```